src/codegeneration/ES6ClassTransformer.js

```diff
@@ -0,0 +1,270 @@
+import {
+  createBindingIdentifier,
+  createClassDeclaration,
+  createCommaExpression,
+  createEmptyParameterList,
+  createFunctionBody,
+  createIdentifierToken,
+  createImmediatelyInvokedFunctionExpression,
+  createParenExpression,
+  createPropertyNameAssignment,} from './ParseTreeFactory.js';
+import {parseStatement} from './PlaceholderParser.js';
+import {propName} from '../staticsemantics/PropName.js';
+import {prependStatements} from './PrependStatements.js';
+
+// TODO(vicb): comment
+
+export class ES6ClassTransformer extends ParseTreeTransformer {
+  /**
+   * @param {UniqueIdentifierGenerator} identifierGenerator
+   * @param {ErrorReporter} reporter
+   * @param {Options} options
+   */
+  constructor(identifierGenerator, reporter, options) {
+    super();
+    this.options_ = options;
+    this.idGenerator_ = identifierGenerator;
+  }
+
+  transformClassElements_(tree) {
+    let superClass = this.transformAny(tree.superClass);
+
+    let elements = [];
+    let initInstanceVars = [], initStaticVars = [];
+    let constructor;
+
+    tree.elements.forEach((tree) => {
+      let initVars;
+      if (tree.isStatic) {
+        initVars = initStaticVars;
+      } else {
+        initVars = initInstanceVars;
+      }
+
+      switch (tree.type) {
+        case GET_ACCESSOR:
+          let getter = new GetAccessor(tree.location, tree.isSatic, tree.name,
+              tree.typeAnnotation, tree.annotations, tree.body);
+          elements.push(this.transformAny(getter));
+          break;
+
+        case SET_ACCESSOR:
+          let setter = new SetAccessor(tree.location, tree.isSatic, tree.name,
+              tree.parameterList, tree.annotations, tree.body);
+          elements.push(this.transformAny(setter));
+          break;
+
+        case PROPERTY_METHOD_ASSIGNMENT:
+          if (!tree.isStatic && propName(tree) === CONSTRUCTOR) {
+            constructor = tree;
+          } else {
+            let pma = new PropertyMethodAssignment(tree.location, tree.isStatic,
+                tree.functionKind, tree.name, tree.parameterList,
+                tree.typeAnnotation, tree.annotations, tree.body,
+                tree.debugName);
+            elements.push(this.transformAny(pma));
+          }
+          break;
+
+        case PROPERTY_VARIABLE_DECLARATION:
+          this.transformAny(tree);
+          if (tree.initializer !== null) {
+            initVars.push(tree);
+          }
+          break;
+
+        default:
+          throw new Error(`Unexpected class element: ${tree.type}`);
+      }
+    });
+
+    let initStatements = getInstanceInitStatements(initInstanceVars);
+
+    if (!constructor) {
+      constructor = this.getDefaultConstructor_(tree, initStatements);
+    } else {
+      if (this.options_.memberVariables) {
+        constructor = this.appendInstanceInitializers_(constructor,
+            initStatements, tree.superClass);
+      }
+    }
+
+    elements.push(constructor);
```

Comment:
Doesn't this put the constructor last? I would prefer if the order is kept and if we have to generated a constructor that it goes first.